allenact/main.py

```diff
@@ -190,6 +190,24 @@ def get_argument_parser():
     )
     parser.set_defaults(disable_config_saving=False)
 
+    parser.add_argument(
+        "--collect_valid_results",
+        dest="collect_valid_results",
+        action="store_true",
+        required=False,
+        help="enables returning and saving valid results during training",
+    )
+    parser.set_defaults(collect_valid_results=False)
+
+    parser.add_argument(
+        "--inference_expert",
+        dest="inference_expert",
+        action="store_true",
+        required=False,
+        help="use expert during valid or test",
+    )
+    parser.set_defaults(inference_expert=False)
```

Comment:
I like this option but don't think we should allow it during validation and, during testing, we should hijack the tag to ensure it's super obvious that the expert is being used.

Reply:
I will block it for validation - I totally see your point. I'll check how to enforce a tag in runner (it shouldn't be too hard, I think).